### PR TITLE
give quicker feedback when setup incorrect

### PR DESCRIPTION
**Pull request by blundell**
Was using on a pet project this weekend and wanted to sync once I added the dependency (but not written the closure yet). You get a cryptic error message like this:

screen shot 2018-02-04 at 7 46 36 pm

So now if you don't have the closure, ie:

screen shot 2018-02-04 at 7 46 49 pm

You get this error message:

screen shot 2018-02-04 at 7 47 03 pm

Also added a change of error message for each missing individual required field of the closure. 

ArtifactId missing:

screen shot 2018-02-04 at 7 47 19 pm

gives:

screen shot 2018-02-04 at 7 47 14 pm

_______

The readme says

> If you use Kotlin DSL use: (do something else)

I Have not tested this in a kotlin prj :-)

### CODE REVIEW

File: core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy [hunk 1]
```diff
@@ -4,20 +4,48 @@ import com.jfrog.bintray.gradle.BintrayPlugin
...
+        }
+        if (extension.publishVersion == null) {
+            extensionError += "Missing publishVersion. "
+        }
+        if (extension.desc == null) {
+            extensionError += "Missing desc. "
+        }
+        if (extensionError) {
+            String prefix = "Have you created the publish closure? "
+            throw new IllegalStateException(prefix + extensionError)
```
Comment (tasomaniac) [inline core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy]: Are they all mandatory? I was thinking the description can be optional.

**Comment by tasomaniac** [comment]
This fixes #118 right?

**Comment by blundell** [comment]
yeah it will give better feedback in that error case

File: core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy (hunk 1 above)
Comment (blundell) [reply core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy]: yes all mandatory

description is mandatory https://github.com/novoda/bintray-release/blob/master/core/src/main/groovy/com/novoda/gradle/release/PublishExtension.groovy#L31

**Pull request merged**